extensions/transform/org.eclipse.smarthome.transform.regex/src/main/java/org/eclipse/smarthome/transform/regex/internal/RegExTransformationService.java

```diff
@@ -66,7 +69,7 @@ public String transform(String regExpression, String source) throws Transformati
             logger.debug(
                     "the given regex '^{}$' doesn't match the given content '{}' -> couldn't compute transformation",
                     regExpression, source);
-            return null;
+            return UnDefType.NULL.toFullString();
```

Comment:
As I said in the commit message: I have aligned it with the other transformation services.

Nevertheless we should mark this PR as API-breaking because people might have used it in rules and now have to test against "NULL" instead of `null`.